docs/admin/notifications.md

```diff
@@ -0,0 +1,251 @@
+# Notifications
+
+Notifications are sent by Coder in response to specific internal events, such as a workspace being deleted or a user
+being created.
+
+**Notifications are currently an experimental feature.**
```

Comment:
This is causing CI to fail since the release is not yet present: https://github.com/coder/coder/actions/runs/10629250397/job/29465701530?pr=14478

We can ignore this failure and merge, or wait to merge until the release has been prepared.

Reply:
We need it in main before we cut the tag for offline docs, let's just skip the link.

Reply:
OK, removed in https://github.com/coder/coder/pull/14478/commits/6a6b903288e757cee5fb2d07c3e2db7ba72824b2.